blocks/rich-text/index.js

```diff
@@ -406,36 +406,6 @@ export class RichText extends Component {
 		this.context.onCreateUndoLevel();
 	}
 
-	/**
-	 * Determines the DOM rectangle for the selection in the editor.
-	 *
-	 * @return {DOMRect} The DOMRect based on the selection in the editor.
-	 */
-	getEditorSelectionRect() {
-		let range = this.editor.selection.getRng();
-
-		// getBoundingClientRect doesn't work in Safari when range is collapsed
-		if ( range.collapsed ) {
-			const { startContainer, startOffset } = range;
-			range = document.createRange();
-
-			if ( ( ! startContainer.nodeValue ) || startContainer.nodeValue.length === 0 ) {
-				// container has no text content, select node (empty block)
-				range.selectNode( startContainer );
-			} else if ( startOffset === startContainer.nodeValue.length ) {
-				// at end of text content, select last character
-				range.setStart( startContainer, startContainer.nodeValue.length - 1 );
```

Comment:
@aduth So this function was manually selecting a the closed character and then calculating the rectangle on that instead of using `range.getClientRects()`. I think the person who added it was trying `range.getBoundingClientRect()` which wouldn't work on collapsed ranges (I guess the "sum" of the rectangles equals zero in `range.getBoundingClientRect()`). Also, this old implementation might be buggy when the caret in right at the end or start of a line.

I added a comment where I try to explain this briefly:

```js
// For normal collapsed ranges, the trick is to use the first (and only)
// untouched client rect from `getClientRects`.
```

Reply:
In other words: `range.getBoundingClientRect()` = sum of `range.getClientRects()` where for a collapsed range this would be one item with a zero width.

Reply:
... and the "bounding" rectangle would end up being nothing, so we have to look at the source rectangle.

Reply:
I hope this makes all sense... I can try to update the comments.